p2p/src/serv.rs

```diff
@@ -147,6 +147,16 @@ impl Server {
 					&self.handshake,
 					self.peers.clone(),
 				)?;
+
```

Comment:
You can probably get rid of all this now?

Reply:
OK